aten/src/TH/generic/THStorage.cpp

```diff
@@ -6,7 +6,7 @@
 
 real* THStorage_(data)(const THStorage *self)
 {
-  return self->data;
+  return self->data<real>();
```

Comment:
Is there any performance impact of this change? This adds runtime type checks where we used to have just a pointer returned.

Reply:
There may be a small impact but:
1) It wasn't big enough to be picked up by the performance regression suite
2) The fast paths and inner loops avoid the runtime check by calling unsafeData
3) We can make it DEBUG only if necessary
4) We are double checking because both ATen and TH check; presumably if it's necessary we can remove the ATen check now that TH is safe.